src/librustc_const_eval/_match.rs

```diff
@@ -561,19 +561,25 @@ fn max_slice_length<'p, 'a: 'p, 'tcx: 'a, I>(
 ///   (1) all_constructors will only return constructors that are statically
 ///       possible. eg. it will only return Ok for Result<T, !>
 ///
-/// Whether a vector `v` of patterns is 'useful' in relation to a set of such
-/// vectors `m` is defined as there being a set of inputs that will match `v`
-/// but not any of the sets in `m`.
+/// This finds whether a (row) vector `v` of patterns is 'useful' in relation
+/// to a set of such vectors `m` is defined as there being a set of inputs
+/// that will match `v` but not any of the sets in `m`.
```

Comment:
This sentence is not 100% grammatical. Maybe:

"This finds whether a (row) vector `v` of patterns is 'useful' in relation to a set of such vectors `m`; useful is defined as there being a set of inputs that will match `v` but not any of the sets in `m`."

Reply:
Also not my fault.